service-api/module/Application/src/Yoti/YotiService.php

```diff
@@ -65,21 +68,49 @@ public function postOfficeBranch(string $postCode): array
      * @param array $sessionData
      * @return array
      * Create a IBV session with applicant data and requirements
+     * is the endpoint there meant to be relative or a full path?
+     * @throws YotiException
      */
     public function createSession(array $sessionData): array
     {
-        //need to use the RequestSigner for the signature here that is on another branch
+        $sdkId = new AwsSecret('yoti/sdk-client-id');
+
+        $body = json_encode($sessionData);
+        $nonce = strval(Uuid::uuid4());
+        $dateTime = new DateTime();
+        $timestamp = $dateTime->getTimestamp();
+        try {
+            $requestSignature = RequestSigner::generateSignature(
+                '/sessions?sdkId=' . $sdkId . '&nonce=' . $nonce . '&timestamp=' . $timestamp,
+                'POST',
+                new AwsSecret('yoti/certificate'),
+                $body
+            );
+        } catch (Http\Exception\YotiAuthException $e) {
+            $this->logger->error('Yoti Authentication problem for session-create [' . $e->getMessage() . '] ', [
+                'data' => [ 'sessionBody' => $body]
+            ]);
```

Comment:
Sorry, previous comment was unclear here: we still need to throw an error to stop it trying to make a request without a signature. I think it's not worth try..catching here and just letting the error bubble up the function above.

Also a general logging point: `$body` contains personal information (name, address, DOB). We try to avoid logging personal info because it becomes a potential security risk.

Reply:
Ah good point